crates/wasm_demo/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+#![cfg(target_arch = "wasm32")]
+#![allow(non_snake_case)]
+
+use ra_ide_api::{Analysis, FileId, FilePosition, FileRange, LineCol, Severity};
+use ra_syntax::{SyntaxKind, TextRange};
+use wasm_bindgen::prelude::*;
+
+mod return_types;
+use return_types::*;
+
+#[wasm_bindgen(start)]
+pub fn start() {
+    console_error_panic_hook::set_once();
+    console_log::init_with_level(log::Level::Warn).expect("could not install logging hook");
+    log::info!("worker initialized")
+}
+
+#[wasm_bindgen]
+pub struct WorldState {
+    analysis: Analysis,
+    file_id: FileId,
+}
+
+#[wasm_bindgen]
+impl WorldState {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        let (analysis, file_id) = Analysis::from_single_file("".to_owned());
+        Self { analysis, file_id }
+    }
+
+    pub fn update(&mut self, code: String) -> JsValue {
+        // TODO: how to update analyisis source?
```

Comment:
You'll need `AnalysisHost` and `AnalysisChange` for this, but re-creating on every change is surely fine for the first stab!